### PR TITLE
DOCS-#85: Add docs on running unidist with different backends

### LINKED ISSUE #85: Add documentation on using/running unidist with different backends
We should add sections/pages on how to run:

* unidist_on_ray
* unidist_on_dask
* unidist_on_mpi
* unidist_on_multiprocessing
* unidist_on_python

using different ways of setting configuration values (CLI, config API, env vars).

### PR DESCRIPTION
**Pull request by YarShev**
Signed-off-by: Igoshev, Yaroslav <yaroslav.igoshev@intel.com>

## What do these changes do?

- [x] passes `flake8 .`
- [x] passes `black .`
- [x] signed commit with `git commit -s` 
- [x] Resolves #85  
- [x] tests passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: docs/using_unidist/index.rst [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+This page contains information on how to run unidist on different execution backends.
+
+- :doc:`unidist on Ray <unidist_on_ray>`
+- :doc:`unidist on MPI <unidist_on_mpi>`
+- :doc:`unidist on Dask <unidist_on_dask>`
+- :doc:`unidist on Python Multiprocessing <unidist_on_python_multiprocessing>`
+- :doc:`unidist on Python <unidist_on_python>`
+
+.. _directory-tree:
```
Comment (prutskov) [inline docs/using_unidist/index.rst:17]: Why is this needed?

File: docs/using_unidist/unidist_on_dask.rst [hunk 2]
```diff
@@ -0,0 +1,37 @@
+..
+      Copyright (C) 2021-2022 Modin authors
+
+      SPDX-License-Identifier: Apache-2.0
+
+:orphan:
+
+unidist on Dask
```
Comment (prutskov) [inline docs/using_unidist/unidist_on_dask.rst:8]: Here, in the title you use `unidist` from small letter, but in using_unidist/index.rts in the title `Using Unidist` with capital letter. Possible, we should align usage of capital letter in the titles and small in other cases (applicable for all files).

File: docs/using_unidist/unidist_on_dask.rst [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+Second, setting the environment variable:
+
+.. code-block:: bash
+
+    # unidist will use Dask backend to distribute computations
+    export UNIDIST_BACKEND=dask
+
+For more information on the environment variables and associated configs specific to the Dask backend
+see :doc:`config API </flow/unidist/config>` section.
```
Comment (prutskov) [inline docs/using_unidist/unidist_on_dask.rst:32]: Is it possible to make a link on specific subtitle from this section? (I mean the link to page https://unidist--140.org.readthedocs.build/en/140/flow/unidist/config.html#unidist-configuration-settings-list) (applicable for all files)

File: docs/using_unidist/unidist_on_mpi.rst [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+.. code-block:: bash
+
+    # unidist will use MPI backend to distribute computations
+    export UNIDIST_BACKEND=mpi
+
+For more information on the environment variables and associated configs specific to the MPI backend
+see :doc:`config API </flow/unidist/config>` section.
+
+.. note::
+   Note that the config ``Backend`` object associated with the ``UNIDIST_BACKEND`` environment variable
```
Comment (prutskov) [inline docs/using_unidist/unidist_on_mpi.rst:35]: The same we should say about `CpuCount`. If we specified the value `UNIDIST_CPUS` or via CLI option the needed amount of workers will be created on finish stage of CLI layer execution. The setting of this value in code using `cfg.CpuCount` adds unexpected effects.

File: docs/using_unidist/index.rst (hunk 1 above)
Comment (YarShev) [reply docs/using_unidist/index.rst:17]: No need, removed

File: docs/using_unidist/unidist_on_dask.rst (hunk 2 above)
Comment (YarShev) [reply docs/using_unidist/unidist_on_dask.rst:8]: aligned

File: docs/using_unidist/unidist_on_dask.rst (hunk 3 above)
Comment (YarShev) [reply docs/using_unidist/unidist_on_dask.rst:32]: I think we can leave as is

File: docs/using_unidist/unidist_on_mpi.rst (hunk 4 above)
Comment (YarShev) [reply docs/using_unidist/unidist_on_mpi.rst:35]: done
